store/src/main/java/com/nytimes/android/external/store2/base/impl/RealInternalStore.java

```diff
@@ -171,24 +173,37 @@ private void initMemCache(MemoryPolicy memoryPolicy) {
      */
     @Nonnull
     @Override
-    public Observable<Parsed> disk(@Nonnull final Key key) {
+    public Maybe<Parsed> disk(@Nonnull final Key key) {
         if (StoreUtil.shouldReturnNetworkBeforeStale(persister, stalePolicy, key)) {
-            return Observable.empty();
+            return Maybe.empty();
         }
 
         return readDisk(key);
     }
 
-    Observable<Parsed> readDisk(@Nonnull final Key key) {
+    Maybe<Parsed> readDisk(@Nonnull final Key key) {
+        return readDisk(key, null);
+    }
+
+    Maybe<Parsed> readDisk(@Nonnull final Key key, final Throwable error) {
         return persister().read(key)
-                .onErrorResumeNext(new OnErrorResumeWithEmpty<Raw>())
+                .onErrorResumeNext(new Function<Throwable, MaybeSource<? extends Raw>>() {
```

Comment:
Here's the part we need to either have Persisters return an error when  empty or have some other operator handle the empty maybe case when a throwable is passed into read disk.